### PR TITLE
Conditional field testcases

### PR DESCRIPTION
**Pull request by cibelius**
## Description

Add 3 test cases:
* Change birth declaration informant form mother to father on review, ensure that father details work correctly
* In correction, mark father details as not available, and ensure that the data persist when fields are brought back
* Correct informant from mother to father, ensure that father details work correctly

## Checklist

- [x] I have linked the correct Github issue under "Development"
- [x] I have tested the changes locally, and written appropriate tests
- [x] I have tested beyond the happy path (e.g. edge cases, failure paths)
- [ ] I have updated the changelog with this change (if applicable)
- [x] I have updated the GitHub issue status accordingly

### CODE REVIEW

File: e2e/testcases/v2-birth/declarations/change-informant-on-review.spec.ts [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+      await loginToV2(page, CREDENTIALS.REGISTRATION_AGENT)
+      await page.click('#header-new-event')
+      await page.getByLabel('Birth').click()
+      await page.getByRole('button', { name: 'Continue' }).click()
+      await page.getByRole('button', { name: 'Continue' }).click()
+    })
+
+    test('Fill child details', async () => {
+      await page.locator('#firstname').fill(declaration.child.name.firstNames)
+      await page.locator('#firstname').blur()
```
Comment (makelicious) [inline e2e/testcases/v2-birth/declarations/change-informant-on-review.spec.ts]: Is the blur needed?

File: e2e/testcases/v2-correction-birth/correct-birth-record-change-informant.spec.ts [hunk 2]
```diff
@@ -0,0 +1,402 @@
...
+      'Reason for correction',
+      'Informant provided incorrect information (Material error)'
+    )
+    await visible(page, 'Fee total', `$${correctionFee}`)
+
+    await visible(page, 'Correction(s)')
+
+    await page
+      .getByRole('heading', { name: 'Record corrected' })
+      .locator('xpath=following-sibling::*[1]')
```
Comment (makelicious) [inline e2e/testcases/v2-correction-birth/correct-birth-record-change-informant.spec.ts]: is this needed?

File: e2e/testcases/v2-correction-birth/correct-birth-record-change-informant.spec.ts [hunk 3]
```diff
@@ -0,0 +1,402 @@
...
+    await expect(
+      page
+        .locator('#listTable-task-history')
+        .getByRole('button', { name: 'Record corrected' })
+    ).toBeVisible()
+  })
+
+  test('Validate record corrected modal', async () => {
+    const correctionRequestedRow = page.locator(
+      '#listTable-task-history #row_7'
```
Comment (makelicious) [inline e2e/testcases/v2-correction-birth/correct-birth-record-change-informant.spec.ts]: is there a way to get by text?

**Review by makelicious — APPROVED** [review]
